doc/html/pcre2unicode.html

```diff
@@ -157,6 +157,42 @@ <h1>pcre2unicode man page</h1>
 counterparts can be disabled by setting the PCRE2_EXTRA_CASELESS_RESTRICT
 option. When this is set, all characters in a case equivalence must either be
 ASCII or non-ASCII; there can be no mixing.
+<pre>
+    Without PCRE2_EXTRA_CASELESS_RESTRICT:
+      'k' = 'K' = U+212A (Kelvin sign)
+      's' = 'S' = U+017F (long S)
+    With PCRE2_EXTRA_CASELESS_RESTRICT:
+      'k' = 'K'
+      U+212A (Kelvin sign)  only case-equivalent to itself
+      's' = 'S'
+      U+017F (long S)       only case-equivalent to itself
+</PRE>
+</P>
+<P>
+One language family, Turkish and Azeri, has its own case-insensitivity rules,
+which can be selected by setting PCRE2_EXTRA_TURKISH_CASING. This alters the
+behaviour of the 'i', 'I', U+0130 (capital I with dot above), and U+0131
+(small dotless i) characters.
+<pre>
+    Without PCRE2_EXTRA_TURKISH_CASING:
+      'i' = 'I'
+      U+0130 (capital I with dot above)  only case-equivalent to itself
+      U+0131 (small dotless i)           only case-equivalent to itself
+    With PCRE2_EXTRA_TURKISH_CASING:
+      'i' = U+0130 (capital I with dot above)
+      U+0131 (small dotless i) = 'I'
+</PRE>
+</P>
+<P>
+It is allowed to specify both PCRE2_EXTRA_CASELESS_RESTRICT and
+PCRE2_EXTRA_TURKISH_CASING together. The behaviour is to apply
+PCRE2_EXTRA_CASELESS_RESTRICT first, removing case-folding mappings which
+cross the ASCII boundary; and then to apply PCRE2_EXTRA_TURKISH_CASING to add
+the Turkish case-folding mappings for 'i'. The result is that there are two
+case-folding mappings crossing the ASCII boundary (the Turkish 'i' ones, as
+requested), but the other mappings (Kelvin sign and long S) are removed.
+If you require exact ASCII matching of ASCII characters in patterns, then
+you must use PCRE2_EXTRA_CASELESS_RESTRICT without PCRE2_EXTRA_TURKISH_CASING.
```

Comment:
You and Carlo both think that it's confusion. So maybe we could just forbid them from being used together.

That's not easy though... since the `(?r)` flag can be set mid-pattern!

If you use Turkish, you really do need the case mappings that cross the ASCII boundary. It would be broken to have some half-way behaviour, that disassociates the 'i' and 'I' characters, but doesn't add the association between those characters and U+130, U+131.

Reply:
I support the mutual exclusivity. I think it was even suggested before. If somebody really needs it in the future, we can think about that. And hear his use case first...

Reply:
> If you use Turkish, you really do need the case mappings that cross the ASCII boundary.

Since we are going to produce an error, we should also (at least for now) trigger it when the utf/ucp flags aren't also provided, because `turkish_caseless` ONLY knows how to make that association in that case.

if an 8-bit codepage is in use, is also not clear what `turkish_caseless` should do.

Reply:
What should be the behaviour be if you do `/...(?r: foo ).../i,turkish_casing`? Should it deactivate the turkish casing, when ASCII-restrict is activated, so i and I become equal again? Or just "break" the turkish casing, leaving all 4 of the 'i' characters not equal to each other?

They're both nasty.

Reply:
I don't mind as long as the behaviour is clearly documented. The /...(?r:foo).../i,turkish_casing example is a plausible scenario for a pattern that is looking for something ascii in the middle of some Turkish, in which case turning off turkish casing does make sense. I think this is, in fact, what Perl does. With a hacked version of perltest.sh (I will tidy this up and commit in due course) I get this:

```
Locale: tr_TR.utf8
Perl v5.40.0
/\x{130}/gi,utf
    \x{130}\x{131}iI
 0: \x{130}
 0: i

/\x{131}/gi,utf
    \x{130}\x{131}iI
 0: \x{131}
 0: I

/(?aa)\x{130}/gi,utf
    \x{130}\x{131}iI
 0: \x{130}

/(?aa)\x{131}/gi,utf
    \x{130}\x{131}iI
 0: \x{131}

```
In other words, U+0130 matches i and U+0131 matches I in Turkish \i mode, but if ascii is forced, they don't.

Reply:
I prefer to return a compiler error

Reply:
A compiler error would not be compatible with Perl, and there is a plausible use case for what Perl does.